src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs

```diff
@@ -0,0 +1,256 @@
+                    viewSchedule.Name = name;
+                    break;
+            }
+
+            TransactionManager.Instance.TransactionTaskDone();
+
+            return viewSchedule;
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        ///     Create Schedule by Category, Type and Name.
+        /// </summary>
+        /// <param name="category"></param>
+        /// <param name="name"></param>
+        /// <param name="scheduleType"></param>
+        /// <returns name="scheduleView">Schedule</returns>
+        public static ScheduleView CreateSchedule(Category category, string name, string scheduleType)
+        {
+            if (category == null || name == null)
+            {
+                throw new System.ArgumentNullException("Category or Name");
+            }
+
+            ScheduleType t = (ScheduleType)System.Enum.Parse(typeof(ScheduleType), scheduleType);
+            return new ScheduleView(category, name, t);
+        }
+
+        ///// <summary>
+        /////     Remove Schedule Field from Schedule.
+        ///// </summary>
+        ///// <param name="field"></param>
+        ///// <returns name="View">View Schedule</returns>
+        //public ScheduleView RemoveField(Revit.Schedule.ScheduleField field)
+        //{
+        //    if (this.Fields.Any(x => x.Name == field.Name))
+        //    {
+        //        TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+        //        this.InternalViewSchedule.Definition.RemoveField(field.InternalScheduleField.FieldId);
+        //        TransactionManager.Instance.TransactionTaskDone();
+        //    }
+
+        //    return this;
+        //}
+
+        /// <summary>
+        ///     Add Field.
+        /// </summary>
+        /// <param name="field"></param>
+        /// <returns></returns>
+        public ScheduleView AddField(Revit.Schedule.SchedulableField field)
+        {
+            if (!this.Fields.Any(x => x.Name == field.Name))
+            {
+                TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+                this.InternalViewSchedule.Definition.AddField(field.InternalSchedulableField);
+                TransactionManager.Instance.TransactionTaskDone();
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        ///     Schedule Fields.
+        /// </summary>
+        public List<Revit.Schedule.ScheduleField> Fields
+        {
+            get
+            {
+                IList<Autodesk.Revit.DB.ScheduleFieldId> fieldIds = this.InternalViewSchedule.Definition.GetFieldOrder();
+                List<Revit.Schedule.ScheduleField> fields = new List<Schedule.ScheduleField>();
+                if (fieldIds.Count > 0)
+                {
+                    foreach (Autodesk.Revit.DB.ScheduleFieldId id in fieldIds)
```

Comment:
I think this can be shortened to:
```c#
return fieldIds.Select(id=>new Revit.Schedules.ScheduleField(this.InternalViewSchedule.Definition.GetField(id));
```